src/nvim/file_search.c

```diff
@@ -491,10 +491,13 @@ vim_findfile_init (
       }
 
       if (search_ctx->ffsc_wc_path != NULL) {
+        const size_t wc_path_size = STRLEN(search_ctx->ffsc_wc_path) + 1
+            + STRLEN(search_ctx->ffsc_fix_path + len);
+        if (wc_path_size >= MAXPATHL) {
+          goto error_return;
+        }
```

Comment:
I added this check because xmalloc cannot fail. It shouldn't make a difference for the new test added but I don't know if `wc_path_size` is guaranteed to be less than `MAXPATHL`.